lib/deploy/eb/application.rb

```diff
@@ -0,0 +1,27 @@
+module Eb
+  class Application
+    def initialize(name)
+      @name = name
+    end
+
+    def environments
+      request = {application_name: @name}
+      response = elasticbeanstalk.describe_environments(request)
+      response.environments.map(&:environment_name)
+    end
+
+    private
+
+    def elasticbeanstalk
+      @elasticbeanstalk ||=
+        call_with_error_handling { Aws::ElasticBeanstalk::Client.new }
+    end
+
+    def call_with_error_handling
```

Comment:
I tend to wonder, how many times have we duplicated this method? Maybe time to extract it into our own wrapper?